borg/archiver.py

```diff
@@ -823,6 +811,50 @@ def do_upgrade(self, args):
             print("warning: %s" % e)
         return self.exit_code
 
+    @with_repository(cache=True, exclusive=True)
+    def do_recreate(self, args, repository, manifest, key, cache):
+        """Re-create archives"""
+        def interrupt(signal_num, stack_frame):
+            if recreater.interrupt:
+                print("Received signal, again. I'm not deaf.\n", file=sys.stderr)
+            else:
+                print("Received signal, will exit cleanly.\n", file=sys.stderr)
```

Comment:
Ah, they're on the wrong end. The LF must be before the message: If a progress line is around this will force this status message into the next line. Otherwise it will just partially overwrite the progress line.